msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java

```diff
@@ -0,0 +1,143 @@
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.b2c.B2CPromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.b2c.B2CProvider;
+import com.microsoft.identity.client.ui.automation.interaction.b2c.IdLabB2cSisoPolicyPromptHandler;
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+public class B2CIdLabSisoPolicyTestClone extends AbstractB2CTest {
+
+    final static B2CProvider[] b2CProviders = new B2CProvider[]{
+            B2CProvider.Local,
+            B2CProvider.MSA,
+            B2CProvider.Google,
+            B2CProvider.Facebook,
+    };
+
+    @Parameterized.Parameters(name = "{0}")
+    public static B2CProvider[] data() {
+        return b2CProviders;
+    }
+
+    private final B2CProvider mB2cProvider;
+
+    public B2CIdLabSisoPolicyTestClone(@NonNull final B2CProvider b2CProvider) {
+        mB2cProvider = b2CProvider;
+    }
+
+    @Override
+    public B2CProvider getB2cProvider() {
+        return mB2cProvider;
+    }
+
+    public void testCanLoginWithLocalAndSocialAccounts() throws Throwable {
+        final String username = mLoginHint;
+        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(mLoginHint)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                ((IApp) mBrowser).handleFirstRun();
+
+                final B2CPromptHandlerParameters promptHandlerParameters = B2CPromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(mLoginHint)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(false)
+                        .broker(null)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .b2cProvider(getB2cProvider())
+                        .build();
+
+                new IdLabB2cSisoPolicyPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        },TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        // ------ do silent request ------
+
+        IAccount account = getAccount();
```

Comment:
I don't think this will give you the account now that the wrappers are being used. I suggest you make the `getAccount` method in `MsalSdk` `public` and then use that here

Reply:
updated.